### DISCUSSION TITLE
How do I connect a link to an element at its midpoint?

### DISCUSSION THREAD

**Discussion by reshin-pulapaddy** [clientIO/joint#2298] [Q&A]
### Introduction

How do I connect a link to an element at its midpoint?
![d](https://github.com/clientIO/joint/assets/117301232/f179d9b7-1faf-45a1-abeb-d33c69fca52e)

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment]
You need to set the [connectionPoint](https://resources.jointjs.com/docs/jointjs/v3.7/joint.html#connectionPoints.intro) to be equal to the [anchor](https://resources.jointjs.com/docs/jointjs/v3.7/joint.html#anchors.intro).

The `anchor` is in the center of the shape / magnet by default:
```js
paper.options.defaultAnchor = { name: 'center' }
```

The default `connectionPoint` tries to find a point between the anchor and the other end of the link that lies on the rectangular boundary of the shape.
```js
paper.options.defaultConnectionPoint = { name: 'rectangle' }
```

If you want the link to end exactly in the middle of the shape, you must tell the link to use the `anchor` as its `connection point`.
```js
paper.options.defaultConnectionPoint = { name: 'anchor' }
```

**Reply by reshin-pulapaddy** [reply]
@kumilingus 
How can I connect the link at the midpoint of the side?

like this

![image](https://github.com/clientIO/joint/assets/117301232/3065caec-dad5-4add-b646-d362c1f8ef8f)

![image](https://github.com/clientIO/joint/assets/117301232/9b8bf869-23d7-4c99-b10d-114583e6285c)

**Reply by kumilingus** [reply]
```js
paper.options.defaultAnchor = { name: 'center' }
paper.options.defaultConnectionPoint = { name: 'boundary' }
```
If needed you can use [magnetSelector](https://resources.jointjs.com/docs/jointjs/v3.7/joint.html#dia.attributes.magnetSelector) and [selector](https://resources.jointjs.com/docs/jointjs/v3.7/joint.html#connectionPoints.boundary) `boundary` connection point option.

**Reply by reshin-pulapaddy** [reply]
@kumilingus 
I used  defaultAnchor as midSide it is working fine it is attaching link  at the midside of the element

`defaultAnchor: { name: 'midSide' },`

![ezgif com-video-to-gif (3)](https://github.com/clientIO/joint/assets/117301232/3f00518b-0624-497a-8b06-7d92fa8b8636)

but in the case of path elements, it is not attached on top of the element 
![2](https://github.com/clientIO/joint/assets/117301232/3635b649-bdc1-4719-aa0b-9c7a3633696c)

I need to connect the link like this
![1](https://github.com/clientIO/joint/assets/117301232/bbdddde6-0f63-421c-a737-e0854e6ddba1)


paper configuration

`const paper = (this.paper = new joint.dia.Paper({
			width: this.backgroundImage.width,
			height: this.backgroundImage.height,
			gridSize: 5,
			drawGrid: true,
			model: graph,
			snapLinksSelf: true,
			restrictTranslate: true,
			cellViewNamespace: joint.shapes,
			autoResizePaper: false,
			defaultAnchor: { name: 'midSide' },
			defaultConnectionPoint: { name: 'boundary' },
			defaultLink: paperOptions.defaultLink,
			interactive: this.interactive,
			async: true,
			snapLabels: true,
			sorting: joint.dia.Paper.sorting.APPROX,
			validateConnection: paperOptions.validateConnection,
			background: {
				image: this.backgroundImage.dataURL,
			},
		}));`

**Reply by kumilingus** [reply]
You might want to use the `boundary` connection point option: `extrapolate`.

**Reply by reshin-pulapaddy** [reply]
@kumilingus 
I used the above code
When the link is  in an orthogonal router, when I try to connect the link to the top of the element it automatically connects at the left side of the element

![gf](https://github.com/clientIO/joint/assets/117301232/591ad44c-2594-469f-b320-8c2aae4b3b3c)

**Reply by kumilingus** [reply]
Please see https://github.com/clientIO/joint/discussions/2254.
